src/cpl/share_esmf/ExcessIceStreamType.F90

```diff
@@ -0,0 +1,295 @@
+    integer                        :: day                ! day of month (1, ..., 31) for nstep+1
+    integer                        :: sec                ! seconds into current date for nstep+1
+    integer                        :: mcdate             ! Current model date (yyyymmdd)
+    type(shr_strdata_type)         :: sdat_exice         ! input data stream
+    character(len=16), allocatable :: stream_varnames(:) ! array of stream field names
+    integer                        :: rc                 ! error code
+    real(r8), pointer              :: dataptr1d(:)       ! temporary pointer
+    character(len=*), parameter    :: stream_name = 'excess ice'
+
+    call this%InitAllocate( bounds )
+    call control%ReadNML( bounds, NLFileName )
+    if ( this%useStreams() )then
+      allocate(stream_varnames(1))
+            stream_varnames = (/"EXICE"/)
+      
+      if (masterproc) then
+        write(iulog,*) '  stream_varnames                  = ',stream_varnames
+      end if
+
+      call shr_strdata_init_from_inline(sdat_exice,                                    &
+      my_task             = iam,                                                &
+      logunit             = iulog,                                              &
+      compname            = 'LND',                                              &
+      model_clock         = model_clock,                                        &
+      model_mesh          = mesh,                                               &
+      stream_meshfile     = control%stream_meshfile_exice,                      &
+      stream_lev_dimname  = 'null',                                             &
+      stream_mapalgo      = control%stream_mapalgo_exice,                       &
+      stream_filenames    = (/trim(control%stream_fldFileName_exice)/),         &
+      stream_fldlistFile  = stream_varnames,                                    &
+      stream_fldListModel = stream_varnames,                                    &
+      stream_yearFirst    = 1996,                                               &
+      stream_yearLast     = 1996,                                               &
+      stream_yearAlign    = 1,                                                  &
+      stream_offset       = 0,                                                  &
+      stream_taxmode      = 'extend',                                           &
+      stream_dtlimit      = 1.0e30_r8,                                          &
+      stream_tintalgo     = 'nearest',                                          &
+      stream_name         = 'excess ice ',                                      &
+      rc                  = rc)
+      if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) then
+        call ESMF_Finalize(endflag=ESMF_END_ABORT)
+      end if
+      
+      ! Explicitly set current date to a hardcoded constant value. Otherwise
```

Comment:
This is taken from my example of ch4FinundationStreamMod. We should assess if this problem still exists, if not it would be good to remove the hardcoding of the date. We'll talk about this more. I should file an issue on this problem. I'm glad you point it out here, I'm just hoping it's no longer a problem.

Reply:
Should `!TODO` be put here?

Reply:
Yes, we should look into this more, and add a TODO item in the code here.

Reply:
I've filed an issue for this #1820, and am looking into seeing if this is still a problem.

Reply:
This is an issue we'll leave in place, until more work is done on it. I looked at this last year, and I'm pretty sure I remember it still being an issue, so we'll leave it like it is.